Mathlib/Data/Set/Function.lean

```diff
@@ -760,7 +760,23 @@ lemma InjOn.image_inter {s t u : Set α} (hf : u.InjOn f) (hs : s ⊆ u) (ht : t
   exact ⟨y, ⟨ys, zt⟩, hy⟩
 #align set.inj_on.image_inter Set.InjOn.image_inter
 
-end injOn
+lemma InjOn.image (h : s.InjOn f) : s.powerset.InjOn (image f) :=
+  fun s₁ hs₁ s₂ hs₂ h' ↦ by rw [← h.preimage_image_inter hs₁, h', h.preimage_image_inter hs₂]
+
+theorem InjOn.image_eq_image_iff_of_subset (h : s.InjOn f) (h₁ : s₁ ⊆ s) (h₂ : s₂ ⊆ s) :
```

Comment:
for these and the next two lemmas, it's probably unambiguous to remove `_of_subset`?

Reply:
possibly even use `image_inj` instead of `image_eq_image`